woocommerce/woocommerce-gateway-stripe
# Allow customizing the label and description of the UPE payment methods

Fixes #2993 

In legacy experience, merchants can customize the title and description of the payment methods and they are also reflected on the checkout page. This feature was missing from the new checkout experience. In this PR, I have added the ability to customize the title and description of the payment methods when UPE is enabled. 

## Changes proposed in this Pull Request:
- In legacy experience, the settings for each method are saved in their own individual options, like `woocommerce_stripe_ideal_settings`, `woocommerce_stripe_eps_settings` etc.
- For UPE (new checkout experience) I am using the same option to keep the title and description of a payment method. By using the same option, we can ensure the data persists when a merchant moves to the new checkout experience.

## Testing instructions
- Go to the Stripe settings page and disable the legacy experience (enable UPE).
- Check the payment methods list. in `develop` branch, there is no `Customize` button beside the payment methods. In this branch, you should see a `Customize` button beside each payment method.
- Click the `Customize` button of a payment method and confirm that it has a title and description populated which was saved as part of the legacy experience. (This is true for the payment method existing in legacy experience, not for the new methods like klarna or afterpay)
- Change the title and description of a method and click `Save changes`. Reload the page and confirm that the changes persist.

![settings](https://github.com/woocommerce/woocommerce-gateway-stripe/assets/33387139/7ef306d4-0eae-4ce9-9097-6b376b897c88)
 
- Enable a few payment methods.
- As a shopper add a product to the cart and go to the checkout page. Confirm that the payment methods are showing the saved title and description. Check on both shortcode and block checkout pages.

**Shortcode**
![sc title desc](https://github.com/woocommerce/woocommerce-gateway-stripe/assets/33387139/858c85fb-fbbd-4c58-bb6e-53d1604cbdee)

**Block**

![block title desc](https://github.com/woocommerce/woocommerce-gateway-stripe/assets/33387139/cfbaa478-39c4-47f7-9df1-19c09791618e)
